### PR TITLE
Refine the drivers/note Kconfig

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

- driver/note: Restore SCHED_INSTRUMENTATION config to sched/Konfig 
- drivers/note: Remove the dependence of noteram driver
- drivers/note: Change DRIVER_NOTExxx to DRIVERS_NOTExxx

depends on https://github.com/apache/nuttx-apps/pull/1567

## Impact

drivers/note

## Testing

CI

### CODE REVIEW

**Comment by acassis** [comment]
@xiaoxiang781216 I didn't get the complete idea, why it is needed to move this driver from drivers/note/ just because some boards could have their our implementation. Could just keep the generic implementation inside drivers/note and use the board specific like a CUSTOM driver?

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 I didn't get the complete idea, why it is needed to move this driver from drivers/note/

All note drivers are still in drivers/note, but SCHED_INSTRUMENTATION move back to sched/Kconfig(the initial place) which is moved to drivers/Kconfig by:
https://github.com/apache/nuttx/pull/7986/commits/56c9c1bdb23b2f343f96202d6cf1f82156e3d455

> just because some boards could have their our implementation. Could just keep the generic implementation inside drivers/note and use the board specific like a CUSTOM driver?

Yes, that's the goal of this patch, user could enable SCHED_INSTRUMENTATION, but disable DRIVERS_NOTE, and then implement all sched_note_xxx callback in a CUSTOM driver.

File: sched/Kconfig [hunk 1]
```diff
@@ -945,6 +945,148 @@ config SCHED_CPULOAD_TIMECONSTANT
...
+		Monitor only CPUs in the bitset.  Bit 0=CPU0, Bit1=CPU1, etc.
+
+config SCHED_INSTRUMENTATION_HIRES
+	bool "Use Hi-Res RTC for instrumentation"
+	default n
+	---help---
+		Use higher resolution system timer for instrumentation.
+
+config SCHED_INSTRUMENTATION_FILTER
+	bool "Instrumenation filter"
```
Comment (hartmannathan) [inline sched/Kconfig:983]:
```suggestion
	bool "Instrumentation filter"
```

Comment (xiaoxiang781216) [reply sched/Kconfig]: Done.

**Comment by xiaoxiang781216** [comment]
@acassis do you have more concern?

**Review by anchao — APPROVED** [review]

**Pull request merged**